In the first program, x will be 3 at the end of execution and will be output as such. There should be no error.

In the second, x will again have the final value of 3 but will not be printed because it's out of scope at the "puts x" instruction. An error regarding an unknown variable will result.
